src/main/java/javapoz24/team3/rental/application/RentingService.java

```diff
@@ -51,7 +51,10 @@ public RentingService(RentActDomainService rentActDomainService,
     public ResponseEntity<RentActDTO> addOrUpdate(RentActDTO rentActDTO) {
         Optional<Customer> optCustomer = customerService.getById(rentActDTO.getCustomerId());
         Optional<Employee> optRentingEmployee = employeeService.getEmployeeById(rentActDTO.getRentingEmployeeId());
-        Optional<Employee> optClosingEmployee = employeeService.getEmployeeById(rentActDTO.getClosingEmployeeId());
+        Optional<Employee> optClosingEmployee = Optional.empty();
+        if (rentActDTO.getClosingEmployeeId() != null) {
+            optClosingEmployee = employeeService.getEmployeeById(rentActDTO.getClosingEmployeeId());
+        }
```

Comment:
Pokombinuj tak żeby pozbyć sie ifa i zrobić to funkcyjnie, fajne ćwiczenie będzie

Reply:
Wygląda, że następujący kod działa:
Optional<Employee> optClosingEmployee = Optional.ofNullable(rentActDTO.getClosingEmployeeId())
                .flatMap(employeeService::getEmployeeById);
Sprawdzę to jeszcze przed komitem i mergem, bo co tu się dzieje to do końca nie wiem ;)

Reply:
FlatMap używamy zazwyczaj zeby nie było Optional<Optional<costam>>

Reply:
To jest propozycja InteliJ, przed nią stworzyłem coś z końcówką .orElseGet( ()->null ) (już nie pamiętam)

Reply:
Prawdopodobnie getEmployeeById zwraca Ci Optionala więc przy zwykłym map dostałbyś Optional<Optional<Employee>>, więc tak jak jest moim zdaniem jest poprawnie.